nbgrader/converters/base.py

```diff
@@ -359,3 +363,28 @@ def _handle_failure(gd):
 
             self.log.error(msg)
             raise NbGraderException(msg)
+
+    def recursively_archive(self, directory, archive_name):
+        all_files_and_folders = [ item for item in os.listdir(directory) if item != archive_name+".zip" ]
+
+        for item in all_files_and_folders:
```

Comment:
I would use `os.walk` here rather than first coming up with a list of all the files and folders. You can then use `continue` to skip if the file matches the archive name.